rosetta-source/src/main/rosetta/legaldocumentation-csa-type.rosetta

```diff
@@ -1,8 +1,182 @@
+ -       0.50   HP Pays 0.5     HP Pays 0.5     HP Pays 0.5     HP Pays 0.5 
+ -       (0.50) PP Pays 0.5     No Payment      PP Pays 0.5     No Payment  
+ (1.00)  0.50   PP Pays 1.0     No Payment      PP Pays 0.5     No Payment  
+ (1.00)  (0.50) PP Pays 1.0     No Payment      PP Pays 1.5     No Payment  
+ (1.00)  (1.00) PP Pays 1.0     No Payment      PP Pays 2.0     No Payment  
+ (1.00)  1.00   PP Pays 1.0     No Payment      No Payment      No Payment  
+ (1.00)  1.25   PP Pays 1.0     HP Pays 0.25    HP Pays 0.25    No Payment  
+   
+    */
+
+
+type CollateralInterestHandlingParameters: <"Represents parameters that describe how calculated interest amounts are handled, i.e. are they transferred/distributed, or is the collateral balance adjusted, is netting done, and any other special handling.">
+    interestPaymentHandling CollateralInterestHandlingEnum (1..1) <"Specifies how the collateral interest is to be handled.">
+    paymentBusinessCenter BusinessCenterEnum(0..*) <"Specifies applicable business centers for payments.">
+    netPostedAndHeldInterest boolean(1..1) <"Indicates whether to net Held and Posted Interest Payments (i.e. whether interest payable for a period can be netted with interest receivable).">
+    netInterestWithMarginCalls boolean (1..1) <"Indicates whether the interest amount may be offset against any margin call deliver or return amounts?   (aka 'net payments' indicator).">
+    includeAccrualInMarginCalc boolean(1..1) <"Indicates whether or not to include the open interest accrual in the margin calculation.">
+    accrueInterestOnUnsettledInterest boolean(0..1) <"Indicates whether interest accruing on unsettled interest amount is included (continues to be accrued) in the following period.">
+    interestRolloverLimit NumberBound(0..1) <"Specifies the level below which the interest will be rolled over.">
+
+    writeoffLimit NumberBound(0..1) <"Specifies the level below which the interest will be written off; if omitted write-off is not applicable.">
+
+    alternativeToInterestAmount AlternativeToInterestAmountEnum (0..1) <"Specifies the alternative to interest amounts.">
+    alternativeProvision string (0..1) <"Specifies an alternative to interest amount, when the alternative provision clause is specified.">
+
+    cutoffTime time(0..1)  <"Specifies the time of day that interest needs to be confirmed by.">
+    notification CollateralInterestNotification(0..1) <"Specifies the terms describing notification requirements.">
+    onFullReturn boolean(1..1) <"Indicates the option that accrued interest should be calculated and distributed when a full return of collateral occurs.">
+    onPartialReturn boolean(1..1) <"Indicates the option that accrued interest should be calculated and distributed when a partial return collateral occurs.">
+
+
+
+type CollateralInterestFrequency: <"Represents the parameters for describing how often collateral interest is to be calculated.">
```

Comment:
Rename and move to `base.datetime`. Also adjust the descriptions to make them generic.